hedera-node/src/main/java/com/hedera/services/state/submerkle/ExpirableTxnRecord.java

```diff
@@ -325,6 +337,9 @@ public void serialize(SerializableDataOutputStream out) throws IOException {
 		out.writeByteArray(alias.toByteArray());
 		out.writeByteArray(ethereumHash);
 		writeNullableSerializable(stakingRewardsPaid, out);
+
+		out.writeInt(pseudoRandomNumber);
```

Comment:
At 10k TPS the state will contain ~1.8M records; so even if none are for `RandomGenerate`, this will add 14.4MB to state.

We can cut that overhead to 1.8MB when no records are for `RandomGenerate` if we add a flag in the stream,
```
if (pseudoRandomNumber >= 0) {
    out.writeBoolean(true);
    out.writeInt(pseudoRandomNumber);
} else if (pseudoRandomBitString.length > 0) {
    out.writeBoolean(true);
    out.writeByteArray(pseudoRandomBitString);
} else {
    out.writeBoolean(false);
}
```

Reply:
Agreed. Fixed !